packages/query-core/src/tests/queryObserver.test.tsx

```diff
@@ -554,7 +550,8 @@ describe('queryObserver', () => {
     // Should not log an error
     queryClient.clear()
     await sleep(40)
-    expect(mockLogger.error).not.toHaveBeenNthCalledWith(1, 'reject 1')
+    expect(consoleMock).not.toHaveBeenNthCalledWith(1, 'reject 1')
+    consoleMock.mockRestore()
```

Comment:
it doesn't make any sense to check this assertion via the logger. We're not logging rejections anymore, so it will always be zero. Previously, there was a case where we wanted to test that, according to the comment:

```
    // A second reject is triggered for the retry
    // but the retryer has already set isResolved to true
    // so it does nothing and no error is thrown
```

as it stands, `consoleMock` will _never_ be called with `reject 1` so the test will always succeed. I'm not sure if we can test this any better or if we should just remove the test.

Reply:
As of now I have removed the consoleMock and expect,
I guess we can take an approach where we are checking if the observer's 'isResolved' is set to true before the retry function is called, this would make sure that it is not attempting to retry if the query is resolved.

Reply:
yeah now the test has no `expect` calls so it will always succeed.

> I guess we can take an approach where we are checking if the observer's 'isResolved' is set to true before the retry function is called, this would make sure that it is not attempting to retry if the query is resolved.

👍

Reply:
not sure about this, but I was not able to check the status of the query as it always comes out to be loading, I tried failureCount which returns the amount of failed tires, and asserted that.

```tsx
    expect(result.failureCount).toBe(1);
    expect(result.failureReason).toBe("reject 1");
```

I guess we can also assert count as it will only be incremented only once.

Reply:
I've just removed the test for now 🤷